### PR TITLE
Ft 51 template engine

### PR DESCRIPTION
**Pull request by MordoJ**
Now templates work properly. This only catch is that links in pages work only on the server, not locally on a device. The only thing left for this task though is to add tests.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -69,13 +69,24 @@
...
       <artifactId>commons-io</artifactId>
       <version>2.11.0</version>
     </dependency>
 
+    <dependency>
+      <groupId>com.github.jknack</groupId>
+      <artifactId>handlebars</artifactId>
+      <version>4.3.0</version>
+    </dependency>
+
```
Comment (clemdiep-heig) [inline pom.xml:89]: I don't think it's necessary to have twice the dependency ?

File: src/main/java/ch/heigvd/statique/utils/Config.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import java.util.Map;
 
 public class Config {
-  private Map<String, Object> config;
+  public Map<String, Object> config;
```
Comment (clemdiep-heig) [inline src/main/java/ch/heigvd/statique/utils/Config.java:7]: Keep it `private`

File: src/main/java/ch/heigvd/statique/utils/Config.java [hunk 3]
```diff
@@ -41,8 +41,9 @@ public Map<String, Object> toRender() {
    * @return The merged config.
    */
   public Config merge(Config other) {
-    Config merged = new Config(this.config);
-    merged.config.putAll(other.config);
-    return merged;
+    HashMap<String, Object> temp = new HashMap<>();
+    temp.put("config", this.config);
+    temp.put("page", other.config);
+    return new Config(temp);
```
Comment (clemdiep-heig) [inline src/main/java/ch/heigvd/statique/utils/Config.java:47]: We have done that to test how `Handlebars.java` works, but it think some rework would be needed for this class, like adding a parameter for the prefixes some where. I will create an issue for that.

File: src/main/java/ch/heigvd/statique/utils/Page.java [hunk 4]
```diff
@@ -5,23 +5,27 @@
...
+  private Template template;
 
   /**
    * Page constructor.
    *
    * @param fromPath The path to the markdown file.
    * @param toPath   The path to the HTML file.
    */
-  public Page(Path fromPath, Path toPath) {
+  public Page(Path fromPath, Path toPath, Template temp) {
```
Comment (clemdiep-heig) [inline src/main/java/ch/heigvd/statique/utils/Page.java:25]: Add the comment for the new parameter

**Review by clemdiep-heig — CHANGES_REQUESTED** [review]
Well, there some big issue with what you commited

1. Remove the directory `site/build`, we don't want it on the git.
2. Keep the test that does not need big rework
3. Correct the `build` command so that it won't copy the `template` folder
4. Correct the `init` command so that it will add a `template` by default

File: pom.xml (hunk 1 above)
Comment (MordoJ) [reply pom.xml:89]: You're right, I don't know how I ended up have it twice.

File: src/main/java/ch/heigvd/statique/utils/Config.java (hunk 2 above)
Comment (MordoJ) [reply src/main/java/ch/heigvd/statique/utils/Config.java:7]: I've changed it back top ✨private✨

File: src/main/java/ch/heigvd/statique/utils/Page.java (hunk 4 above)
Comment (MordoJ) [reply src/main/java/ch/heigvd/statique/utils/Page.java:25]: Now there's a new shiny comment just to describe this parameter 💫

**Comment by MordoJ** [comment]
All requested changes has been applied.

**Review by dylanrodr-heig — APPROVED** [review]

**Pull request merged**